### PR TITLE
Link preview follow up bug fixes

### LINKED ISSUE #19358: Internal link card followup bugfixes
# Problem

Status link previews are not cleared after sending a message, And sending the same status link preview twice doesn't show the preview on the 2nd time

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #19358

### Summary
Fixes some issues with internal link card component, More details on the issue.

status: ready

### CODE REVIEW

File: src/status_im/contexts/chat/messenger/composer/link_preview/events.cljs [hunk 1]
```diff
@@ -38,7 +38,8 @@
       (let [cache      (get-in db [:chat/link-previews :cache])
             previews   (urls->previews cache urls)
             new-urls   (->> previews
-                            (filter :loading?)
+                            (filter (fn [{:keys [loading? status-link-preview?]}]
+                                      (or loading? status-link-preview?)))
```
Comment (ibrkhalil) [inline src/status_im/contexts/chat/messenger/composer/link_preview/events.cljs:42]: Don't count status link previews as cached.

File: src/status_im/contexts/chat/messenger/composer/link_preview/events.cljs [hunk 2]
```diff
@@ -139,14 +140,18 @@
   sent."
   {:events [:link-preview/reset-unfurled]}
   [{:keys [db]}]
-  {:db (update db :chat/link-previews dissoc :unfurled :request-id :cleared)})
+  {:db (-> db
+           (update :chat/link-previews dissoc :unfurled :request-id :cleared)
+           (update :chat/status-link-previews dissoc :unfurled :request-id :cleared))})
```
Comment (ibrkhalil) [inline src/status_im/contexts/chat/messenger/composer/link_preview/events.cljs:145]: Clear status link previews when sending a message.

File: src/status_im/contexts/chat/messenger/composer/link_preview/events.cljs [hunk 3]
```diff
@@ -139,14 +140,18 @@
...
 
 (rf/defn reset-all
   "Reset all preview state. It is especially important to delete any cached
   URLs, as failing to do so results in its unbounded growth."
   {:events [:link-preview/reset-all]}
   [{:keys [db]}]
-  {:db (dissoc db :chat/link-previews)})
+  {:db (-> db
+           (dissoc :chat/link-previews)
+           (dissoc :chat/status-link-previews))})
```
Comment (ibrkhalil) [inline src/status_im/contexts/chat/messenger/composer/link_preview/events.cljs:154]: Clear status link previews when closing a chat.

**Review by Parveshdhull — APPROVED** [review]
Hi @ibrkhalil, Thank you for fixing the issue.

Why we are using the specific key `:chat/status-link-previews` for community previews? Can't we just store them in `:chat/link-previews`?

**Comment by ibrkhalil** [comment]
> Hi @ibrkhalil, Thank you for fixing the issue.
> 
> Why we are using the specific key `:chat/status-link-previews` for community previews? Can't we just store them in `:chat/link-previews`?

I think it's better to have them on a different sub, Because I remember desktop team having them on two separate attributes on the message object.
```clj
{:message-id                "bla bla"
 :link-previews             [...]
 :status-link-previews      [...]}
```
 So it's easier for us to access them directly when sending a message, Instead of having to filter messages.

**Comment by mariia-skrypnyk** [comment]
Hi @ibrkhalil !

Thanks for detecting such issues!
I've checked your fixes on both platforms and they are working really good!
Your PR fixes very tricky and important release bugs.
Please, can I ask you about letting us know that you found something critical for the future release and working on/fixing it.
It would be very informative for QA's and the whole team 🙏 as e.g. we were stuck on the same issue today.

**Comment by ibrkhalil** [comment]
> Hi @ibrkhalil !
> 
> 
> 
> Thanks for detecting such issues!
> 
> I've checked your fixes on both platforms and they are working really good!
> 
> Your PR fixes very tricky and important release bugs.
> 
> Please, can I ask you about letting us know that you found something critical for the future release and working on/fixing it.
> 
> It would be very informative for QA's and the whole team 🙏 as e.g. we were stuck on the same issue today.

Hey @mariia-skrypnyk 
I added this message [here](https://discord.com/channels/1210237582470807632/1217173160416382986/1220489775702609981), But I get QA was too busy with release testing you might have missed it.

**Comment by pavloburykh** [comment]
Hey @ibrkhalil! Thanks for the fix. I am wondering if you have rights to cherrypick the fix into release/2.28.x branch. In case you don't, that's fine, I will ask someone in mobile channel.

**Comment by ibrkhalil** [comment]
> Hey @ibrkhalil! Thanks for the fix. I am wondering if you have rights to cherrypick the fix into release/2.28.x branch. In case you don't, that's fine, I will ask someone in mobile channel.

Unfortunately I don't have access, I'll merge it now.

**Comment by Parveshdhull** [comment]
> > Hey @ibrkhalil! Thanks for the fix. I am wondering if you have rights to cherrypick the fix into release/2.28.x branch. In case you don't, that's fine, I will ask someone in mobile channel.
> 
> Unfortunately I don't have access, I'll merge it now.

done, cherry-picked - https://github.com/status-im/status-mobile/commits/release/2.28.x

**Pull request merged**